FirebasePerformance/Tests/Unit/FPRSessionManagerTest.m

```diff
@@ -91,4 +91,35 @@ - (void)testSessionIdUpdationThrowsNotification {
   XCTAssertNotEqual(sessionId, instance.sessionDetails.sessionId);
 }
 
+/** Validate that sessionId changes sends notifications with the session details. */
+- (void)testSessionIdUpdationSendsNotificationWithSessionDetails {
+  FPRSessionManager *instance = [FPRSessionManager sharedInstance];
+  [instance startTrackingAppStateChanges];
+  NSString *sessionId = instance.sessionDetails.sessionId;
+
+  __block BOOL containsSessionDetails = NO;
+  __block FPRSessionDetails *updatedSessionDetails = nil;
+  [instance.sessionNotificationCenter
+      addObserverForName:kFPRSessionIdUpdatedNotification
+                  object:instance
+                   queue:[NSOperationQueue mainQueue]
+              usingBlock:^(NSNotification *note) {
+                NSDictionary<NSString *, FPRSessionDetails *> *userInfo = note.userInfo;
+                FPRSessionDetails *sessionDetails =
+                    [userInfo valueForKey:kFPRSessionIdNotificationKey];
+                if (sessionDetails != nil) {
+                  containsSessionDetails = YES;
+                  updatedSessionDetails = sessionDetails;
```

Comment:
optional: Currently the test relies on the implementation details so the block happen to be called synchronously with posting the notification. According to the [docs](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1411723-addobserverforname) we may pass `nil` to `queue` to make sure the block is called on the poster thread. Alternatively `XCTestExpectation` can be used to make sure the test works when the block is called asynchronously. Otherwise the test may unexpectedly fail on a different platform or OS version.

Reply:
Sure. That makes sense. We can definitely try Expectation, but expectation also comes with a timeout value.

I would prefer to keep that change (if we decide to do that) outside this PR since that would require multiple tests to be changed as well.